samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Program.cs

```diff
@@ -0,0 +1,139 @@
+if (isLocalEmulator)
+{
+    // For local emulator, no authentication needed
+    connectionString = $"Endpoint={hostAddress};TaskHub={taskHubName};Authentication=None";
+    logger.LogInformation("Using local emulator with no authentication");
+}
+else
+{
+    // For Azure, use DefaultAzure authentication
+    connectionString = $"Endpoint={hostAddress};TaskHub={taskHubName};Authentication=DefaultAzure";
+    logger.LogInformation("Using Azure endpoint with DefaultAzure authentication");
+}
+
+logger.LogInformation("Using endpoint: {Endpoint}", endpoint);
+logger.LogInformation("Using task hub: {TaskHubName}", taskHubName);
+logger.LogInformation("Host address: {HostAddress}", hostAddress);
+logger.LogInformation("Connection string: {ConnectionString}", connectionString);
+logger.LogInformation("This worker implements a news article generator workflow with multiple specialized agents");
+
+// Create loggers for each service and activity
+var researchAgentServiceLogger = loggerFactory.CreateLogger<ResearchAgentService>();
+var contentGenerationAgentServiceLogger = loggerFactory.CreateLogger<ContentGenerationAgentService>();
+var imageGenerationAgentServiceLogger = loggerFactory.CreateLogger<ImageGenerationAgentService>();
+var researchTopicActivityLogger = loggerFactory.CreateLogger<ResearchTopicActivity>();
+var createArticleActivityLogger = loggerFactory.CreateLogger<CreateArticleActivity>();
+var generateImagesActivityLogger = loggerFactory.CreateLogger<GenerateImagesActivity>();
+var assembleFinalArticleActivityLogger = loggerFactory.CreateLogger<AssembleFinalArticleActivity>();
+var orchestrationLogger = loggerFactory.CreateLogger<ContentCreationOrchestration>();
+
+// Create service instances
+var researchAgentService = new ResearchAgentService(researchAgentServiceLogger);
+var contentGenerationAgentService = new ContentGenerationAgentService(contentGenerationAgentServiceLogger);
+var imageGenerationAgentService = new ImageGenerationAgentService(imageGenerationAgentServiceLogger);
+
+// Create activity instances
+var researchTopicActivity = new ResearchTopicActivity(researchAgentService, researchTopicActivityLogger);
+var createArticleActivity = new CreateArticleActivity(contentGenerationAgentService, createArticleActivityLogger);
+var generateImagesActivity = new GenerateImagesActivity(imageGenerationAgentService, generateImagesActivityLogger);
+var assembleFinalArticleActivity = new AssembleFinalArticleActivity(assembleFinalArticleActivityLogger);
+
+// Create orchestration instance
+var contentCreationOrchestration = new ContentCreationOrchestration(orchestrationLogger);
+
+// Configure services
+builder.Services.AddDurableTaskWorker()
+    .AddTasks(registry =>
+    {
+        // Register the orchestration
+        registry.AddOrchestratorFunc<ContentCreationRequest, ContentWorkflowResult>(
+            "ContentCreationOrchestration", 
+            (ctx, input) => contentCreationOrchestration.RunAsync(ctx, input));
```

Comment:
This is not the recommended way to register orchestrations and activities. There's a much simpler way if you instead have your orchestrations derive from `TaskOrchestration<,>` and have your activities derive from `TaskActivity<,>`. See the other samples for an example.